src/snek5000/output/phys_fields.py

```diff
@@ -140,3 +163,151 @@ def change_reader(self, reader_key):
+
+    @lru_cache(maxsize=2)
+    def _get_data_from_path(self, path):
+        return pymech.open_dataset(path)
+
+    def _get_hexadata_from_time(self, time):
+        index = self.times.tolist().index(time)
+        return self._get_hexadata_from_path(self.path_files[index])
+
+    @lru_cache(maxsize=2)
+    def _get_hexadata_from_path(self, path):
+        return pymech.readnek(path)
+
+    def _get_field_to_plot_from_file(self, path_file, key, equation):
+        if equation is not None:
+            raise NotImplementedError
+        hexa_data = self._get_hexadata_from_path(path_file)
+        hexa_field = HexaField(key, hexa_data)
+        return hexa_field, float(hexa_data.time)
+
+    def plot_hexa(self, time, equation=None, percentage_dx_quiver=4.0):
+
+        if equation is not None:
+            raise NotImplementedError
+
+        # temporary hack
+        time = self.times[abs(self.times - time).argmin()]
+
+        hexa_data = self._get_hexadata_from_time(time)
+        fig, ax = plt.subplots()
+
+        xmin, xmax = hexa_data.lims.pos[0]
+        ymin, ymax = hexa_data.lims.pos[1]
+
+        dx_quiver = percentage_dx_quiver / 100 * (xmax - xmin)
+        nx_quiver = int((xmax - xmin) / dx_quiver)
+        ny_quiver = int((ymax - ymin) / dx_quiver)
+
+        x_approx_quiver = np.linspace(dx_quiver, xmax - dx_quiver, nx_quiver)
+        y_approx_quiver = np.linspace(dx_quiver, ymax - dx_quiver, ny_quiver)
+
+        x_quiver = []
+        y_quiver = []
+        vx_quiver = []
+        vy_quiver = []
+
+        # assuming 2d...
+        iz = 0
+
+        for elem in hexa_data.elem:
+            field = elem.temp[0][iz]
+            XX = elem.pos[0][iz]
+            YY = elem.pos[1][iz]
+            x = XX[0]
+            y = YY[:, 0]
+            x_edges = get_edges(x)
+            y_edges = get_edges(y)
+            ax.pcolormesh(x_edges, y_edges, field, shading="flat", vmin=-0.5, vmax=0.5)
+
+            xmin = x.min()
+            xmax = x.max()
+            ymin = y.min()
+            ymax = y.max()
+
+            for y_approx in y_approx_quiver:
+                if y_approx < ymin:
+                    continue
+                if y_approx > ymax:
+                    break
+                iy = abs(y - y_approx).argmin()
+                for x_approx in x_approx_quiver:
+                    if x_approx < xmin:
+                        continue
+                    if x_approx > xmax:
+                        break
+                    ix = abs(x - x_approx).argmin()
+
+                    x_quiver.append(x[ix])
+                    y_quiver.append(y[iy])
+
+                    vx_quiver.append(elem.vel[0, iz, iy, ix])
+                    vy_quiver.append(elem.vel[1, iz, iy, ix])
+
+        ax.quiver(x_quiver, y_quiver, vx_quiver, vy_quiver)
+
+    @staticmethod
+    def time_from_path(path):
+        with open(path, "rb") as file:
+            header = read_header(file)
+        return header.time
```

Comment:
`return read_header(path).time` should do it. I made a recent change which accepts both a file object and a path.